File: app-sdk/src/main/java/code/sdk/util/OkHttpUtil.java
package code.sdk.util;

import java.util.HashMap;
import java.util.concurrent.TimeUnit;

import code.sdk.core.util.TestUtil;
import code.util.MySSLSocketClient;
import code.util.TlsSniSocketFactory;
import code.util.TrueHostnameVerifier;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

public class OkHttpUtil {
    private static HashMap<String, OkHttpClient> mHostOkHttpClient = new HashMap<>();
    private static final int CONNECT_TIMEOUT = 8000;
    private static final int READ_TIMEOUT = 8000;

    public static OkHttpClient getOkHttpClient(String domain) {
        OkHttpClient okHttpClient = mHostOkHttpClient.get(domain);
        if (okHttpClient != null) {
            return okHttpClient;
        }
        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        if (TestUtil.isLoggable()) { //这行必须加 不然默认不打印
            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        } else {
            interceptor.setLevel(HttpLoggingInterceptor.Level.NONE);
        }
        okHttpClient = new OkHttpClient.Builder()
                .addInterceptor(interceptor)
                .sslSocketFactory(new TlsSniSocketFactory(domain), MySSLSocketClient.getX509TrustManager())
                .hostnameVerifier(new TrueHostnameVerifier(domain))
                .connectTimeout(CONNECT_TIMEOUT, TimeUnit.MILLISECONDS)
                .readTimeout(READ_TIMEOUT, TimeUnit.MILLISECONDS)
                .retryOnConnectionFailure(false)
                .build();
        mHostOkHttpClient.put(domain, okHttpClient);
        return okHttpClient;
    }
}

File: app-sdk/src/main/java/code/sdk/util/OkHttpUtil.kt
package code.sdk.util

import code.sdk.core.util.TestUtil
import code.util.MySSLSocketClient.getX509TrustManager
import code.util.TlsSniSocketFactory
import code.util.TrueHostnameVerifier
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import java.util.concurrent.TimeUnit

object OkHttpUtil {
    private val mHostOkHttpClient = HashMap<String, OkHttpClient?>()
    private const val CONNECT_TIMEOUT = 8000
    private const val READ_TIMEOUT = 8000
    fun getOkHttpClient(domain: String): OkHttpClient? {
        var okHttpClient = mHostOkHttpClient[domain]
        if (okHttpClient != null) {
            return okHttpClient
        }
        val interceptor = HttpLoggingInterceptor()
        if (TestUtil.isLoggable()) { //这行必须加 不然默认不打印
            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY)
        } else {
            interceptor.setLevel(HttpLoggingInterceptor.Level.NONE)
        }
        okHttpClient = OkHttpClient.Builder()
            .addInterceptor(interceptor)
            .sslSocketFactory(TlsSniSocketFactory(domain), getX509TrustManager())
            .hostnameVerifier(TrueHostnameVerifier(domain))
            .connectTimeout(CONNECT_TIMEOUT.toLong(), TimeUnit.MILLISECONDS)
            .readTimeout(READ_TIMEOUT.toLong(), TimeUnit.MILLISECONDS)
            .retryOnConnectionFailure(false)
            .build()
        mHostOkHttpClient[domain] = okHttpClient
        return okHttpClient
    }
}
